### PR TITLE
MTD-85, main branch fix + other features

### PR DESCRIPTION
**Pull request by Vobrymn**
Prototype explanation on the simulation page. Needs fine tuning. Some fixes with the border expanding and making the screen scrollable into space off screen (i.e white-space)

### CODE REVIEW

**Review by Vobrymn — COMMENTED** [review]
also made param bar scrollbar hidden, changeable in style.css

File: client/index.html [hunk 1]
```diff
@@ -8,7 +8,7 @@
     />
   </head>
   <body>
-    <div id="app"></div>
+    <div id="app" class="overflow-hidden"></div>
```
Comment (benvarian) [inline client/index.html:11]: what does overflow-hidden change

File: client/src/style.css [hunk 2]
```diff
@@ -26,4 +26,12 @@
     --simulation-color: 245 245 245;
     --navbar-icon-color: 49 47 47;
   }
+
+  .hide-scrollbar {
```
Comment (benvarian) [inline client/src/style.css:30]: can we not get rid off the scrollbar, its an important ui feature for a reason

File: client/src/views/SimulationView.vue [hunk 3]
```diff
@@ -465,7 +470,20 @@ const plotNetwork = (graphData) => {
...
+      <div v-if="graph" id="network-graph" class="relative w-full h-[calc(100vh-36px)]"></div>
+      <div v-else id="sim_explanation" class="relative border rounded p-4 translate-y-[-50px] bg-navbar-primary">
+        <h1 class="font-bold ml-2">Simulation quick start guide:</h1>
+
+        <ul class="list-disc text-sm p-4">
+          <li class="mb-2">Total Nodes: the number of nodes in the simulated network<br>Default:</li>
+          <li class="mb-2">Scheme: the manor in which the simulation will employ MTD strategies<br>Default:</li>
+          <li class="mb-2">MTD Interval: the frequency of applying MTD strategies</li>
+        </ul>
+        WIP: fix after param finalisation. <br>
```
Comment (benvarian) [inline client/src/views/SimulationView.vue:483]: - i would say the param bar is finalised so do the rest now and cleanup the todo + wip comments
- would also merge main in as michael added something to this page and wanna make sure that it blends together well

File: client/src/style.css [hunk 4]
```diff
@@ -27,5 +28,23 @@
     --navbar-color: 183 209 218;
     --simulation-color: 245 245 245;
     --navbar-icon-color: 49 47 47;
+    --complementary-color: 183 230 240;
```
Comment (benvarian) [inline client/src/style.css:31]: please merge main in and use michaels scrollbar, this color combo is terrible

File: client/src/views/SimulationView.vue [hunk 5]
```diff
@@ -524,63 +497,110 @@ const plotNetwork = (graphData) => {
...
+              </li>
+              <li class="mb-2">
+                Total Nodes: the number of nodes in the simulated network<br /><span
+                  class="text-xs"
+                  >Default: 50</span
+                >
+              </li>
+              <li class="mb-2">
+                Total Endpoints: the number of endpoints in the simulated
+                network<br /><span class="text-xs">Default: </span>
```
Comment (benvarian) [inline client/src/views/SimulationView.vue:535]: no default supplied

Comment (benvarian) [reply client/src/views/SimulationView.vue:535]: seed also isnt provided in this lil thing

File: client/src/views/SimulationView.vue [hunk 6]
```diff
@@ -524,63 +497,110 @@ const plotNetwork = (graphData) => {
...
+                network<br /><span class="text-xs">Default: 2</span>
+              </li>
+              <li class="mb-2">
+                Total Layers: the number of layers in the simulated network<br /><span
+                  class="text-xs"
+                  >Default: 4</span
+                >
+              </li>
+              <li class="mb-2">
+                Target Layer: the layer where the target host will be located<br /><span
```
Comment (benvarian) [inline client/src/views/SimulationView.vue:552]: have a default answer for this pls like the other ones

File: client/src/views/SimulationView.vue [hunk 7]
```diff
@@ -524,63 +497,110 @@ const plotNetwork = (graphData) => {
...
+                  class="text-xs"
+                  >Default: 50</span
+                >
+              </li>
+              <li class="mb-2">
+                Total Endpoints: the number of endpoints in the simulated
+                network<br /><span class="text-xs">Default: </span>
+              </li>
+              <li class="mb-2">
+                Total Subnets: the number of sub-networks in the simulated
```
Comment (benvarian) [inline client/src/views/SimulationView.vue:538]: if i am to use the default examples for my values mainly total nodes=50, endpoints=45, the sim wont run cause subnets needs to be over 4 and cant be 5

Comment (benvarian) [reply client/src/views/SimulationView.vue:538]: once these are done ill look again

File: client/src/stores/simulation.js [hunk 8]
```diff
@@ -1,6 +1,7 @@
 import { defineStore } from "pinia";
 import axios from "axios";
-const BACKEND_URL = import.meta.env.VITE_BACKEND_URL;
+// const BACKEND_URL = import.meta.env.VITE_BACKEND_URL;
+const BACKEND_URL = "http://localhost:8001";
```
Comment (benvarian) [inline client/src/stores/simulation.js:4]: please remove this stuff as michael still hasnt supplied his env file after a day so we have to assume he aint got one , so just hardcode it for now like it was before and we can get someone to deal with the rest later

File: client/src/views/SimulationView.vue (hunk 5 above)
Comment (Vobrymn) [reply client/src/views/SimulationView.vue:535]: For Scheme? there is no default, it is required

File: client/src/stores/simulation.js (hunk 8 above)
Comment (Vobrymn) [reply client/src/stores/simulation.js:4]: this is already hard coded, ill just remove the comment

File: server/api/app.py [hunk 9]
```diff
@@ -30,22 +30,20 @@ def strategies():
...
-    strategies = request.json.get("strategies")
+    total_layers = request.json.get("totalLayers")
+    target_layer = request.json.get("targetLayer")
+    seed = request.json.get("seed")
 
     custom_strategies = None
 
-    if not all([mtd_interval, scheme, total_nodes]):
+    if not scheme:
         return {}, 400
```
Comment (nicodeas) [inline server/api/app.py:48]: please dont remove my error handling

File: server/experiments/run.py [hunk 10]
```diff
@@ -419,6 +419,8 @@ def simulate_without_saving(
     :param terminate_compromise_ratio: terminate the simulation if reached compromise ratio
     :param new_network: True: create new snapshots based on network size, False: load snapshots based on network size
     """
+    if total_nodes is None:
+        total_nodes = 50
```
Comment (nicodeas) [inline server/experiments/run.py:423]: remove. Not everthing should be default

**Review by nicodeas — CHANGES_REQUESTED** [review]

File: client/src/components/DropDown.vue [hunk 11]
```diff
@@ -49,11 +50,11 @@ import { mdiArrowDown, mdiArrowUp } from "@mdi/js";
 
 const props = defineProps({
   label: { type: String, default: "" },
-  placeholder: { type: String, default: "Select Scheme" },
-  selected: { type: String, default: "Random" },
+  placeholder: { type: String, default: "" },
+  selected: { type: String, default: "" },
   menuOptions: {
     type: Array,
-    default: () => ["Random", "Simultaneous", "Alternative", "Single", "None"],
```
Comment (benvarian) [inline client/src/components/DropDown.vue:56]: add the defaults back in, this component is legit only gonna get used here so theres no point is specifying params in parent

Comment (benvarian) [reply client/src/components/DropDown.vue:56]: same with the other defaults, no need to pass from parent if theres only one use case for the component

File: client/src/stores/simulation.js [hunk 12]
```diff
@@ -13,25 +13,37 @@ export const useSimulationStore = defineStore("simulation", {
...
+        reqBody.totalLayers = Number(this.parameters.totalLayers);
+      }
+      if (this.parameters.targetLayer) {
+        reqBody.targetLayer = Number(this.parameters.targetLayer);
+      }
+      if (this.parameters.seed) {
+        reqBody.seed = parseInt(this.parameters.seed);
+      }
+
+      console.log(reqBody);
```
Comment (benvarian) [inline client/src/stores/simulation.js:46]: remove console.log

File: client/src/stores/simulation.js [hunk 13]
```diff
@@ -13,25 +13,37 @@ export const useSimulationStore = defineStore("simulation", {
...
-        reqBody.startTime = Number(this.parameters.startTime);
+      if (this.parameters.mtdInterval) {
+        reqBody.mtdInterval = Number(this.parameters.mtdInterval);
       }
       if (this.parameters.finishTime) {
         reqBody.finishTime = Number(this.parameters.finishTime);
       }
-      if (this.parameters.mtdInterval) {
-        reqBody.mtdInterval = Number(this.parameters.mtdInterval);
+      if (this.parameters.totalNodes) {
```
Comment (benvarian) [inline client/src/stores/simulation.js:24]: whats the purpose for all these if statements, all the data should be validated before coming into the store

File: client/src/style.css [hunk 14]
```diff
@@ -19,6 +19,7 @@
     --hover-bg-color: 55, 65, 81;
     --simulation-color: 43 43 43;
     --navbar-icon-color: 164 164 164;
+    --complementary-color: 0 204 153;
```
Comment (benvarian) [inline client/src/style.css:22]: change the name of complementary to something more descriptive as its very open ended as to what it actually does

File: client/src/stores/simulation.js (hunk 13 above)
Comment (Vobrymn) [reply client/src/stores/simulation.js:24]: its not validation, its just checking if they specified a value or not, if not, dont even bother sending a rq value, use backend default

File: client/src/style.css (hunk 14 above)
Comment (Vobrymn) [reply client/src/style.css:22]: button-color

File: client/src/components/DropDown.vue (hunk 11 above)
Comment (Vobrymn) [reply client/src/components/DropDown.vue:56]: ok

File: client/src/stores/simulation.js (hunk 13 above)
Comment (Vobrymn) [reply client/src/stores/simulation.js:24]: Number(null) or Number(str) results in 0 or NaN, not None

File: client/src/stores/simulation.js (hunk 12 above)
Comment (Vobrymn) [reply client/src/stores/simulation.js:46]: done

File: client/src/views/SimulationView.vue [hunk 15]
```diff
@@ -541,63 +511,101 @@ const plotNetwork = (graphData) => {
...
-            totalNodes: '70',
-          }"
-          @apply-scenario="applyPredefinedScenario"
-        />
+    <div class="w-full flex-1 flex flex-col ml-1 mr-1 h-[calc(100vh-36px)]">
+      <div v-if="graph" id="network-graph" class="flex-1 h-[calc(100vh)]"></div>
+      <div v-else class="flex-row">
+        <div
+          id="sim_explanation"
+          class="flex-2 m-10 border rounded p-4 h-[calc(50vh)] overflow-y-auto bg-navbar-primary scrollbar"
```
[the remark this answers is not in the document]
Comment (Vobrymn) [reply client/src/views/SimulationView.vue:519]: done

File: client/src/style.css [hunk 16]
```diff
@@ -19,6 +19,7 @@
     --hover-bg-color: 55, 65, 81;
     --simulation-color: 43 43 43;
     --navbar-icon-color: 164 164 164;
+    --button-color: 0 174 153;
```
Comment (benvarian) [inline client/src/style.css:22]: Can we not use button either, it implies it’s the color for every button, use something like sub color or something along those lines

**Review by nicodeas — APPROVED** [review]

**Pull request merged**